ui-admin/src/portal/images/SiteImageUploadModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ChangeEvent, useRef, useState } from 'react'
+import { Modal } from 'react-bootstrap'
+import Api, { SiteImageMetadata } from 'api/api'
+import { doApiLoad } from 'api/api-utils'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { successNotification } from 'util/notifications'
+import { Store } from 'react-notifications-component'
+import { Button } from 'components/forms/Button'
+import { LoadedPortalContextT } from '../PortalProvider'
+
+/** Renders a modal for an admin to submit a sample collection kit request. */
+export default function SiteImageUploadModal({
+  portalContext,
+  onDismiss, onSubmit,
+  existingImage
+}: {
+    portalContext: LoadedPortalContextT,
+    onDismiss: () => void,
+    existingImage?: SiteImageMetadata,
+    onSubmit: () => void }) {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const [fileName, setFileName] = useState(existingImage?.cleanFileName ?? '')
+
+  const handleFileChange = (newFile: File) => {
+    // only autofill the name field if this is a new image
+    if (!existingImage) {
+      setFileName(cleanFileName(newFile.name))
+    }
+  }
+
+  const { file, FileChooser } = useFileUploadButton(handleFileChange)
+  const uploadImage = () => {
+    if (!file) { return }
+    const version = existingImage?.version ? existingImage.version + 1 : 1
+    doApiLoad(async () => {
+      await Api.uploadPortalImage(portalContext.portal.shortcode, fileName, version, file)
+      Store.addNotification(successNotification('image saved'))
+      onSubmit()
+    }, { setIsLoading })
+  }
+
+  return <Modal show={true} onHide={onDismiss}>
+    <Modal.Header closeButton>
+      <Modal.Title>{existingImage ? 'Update' : 'Upload'} image</Modal.Title>
```

Comment:
This is not working for me. I see "Update image" after choosing "Add image". Perhaps some lingering `existingImage`?

Reply:
yup, great catch -- wasn't ever clearing out the existingImage -- fixed